runtime/runtime/src/balance_checker.rs

```diff
@@ -17,6 +17,99 @@ use near_primitives::version::ProtocolVersion;
 use near_store::{get, get_account, get_postponed_receipt, TrieUpdate};
 use std::collections::HashSet;
 
+/// Returns delayed receipts with given range of indices.
+fn get_delayed_receipts(
+    state: &TrieUpdate,
+    from_index: u64,
+    to_index: u64,
+) -> Result<Vec<Receipt>, StorageError> {
```

Comment:
An alternative is to take an `indexes: Range<u64>`. The `..` would then move to the call-site, making exclusive-range explicit in the API